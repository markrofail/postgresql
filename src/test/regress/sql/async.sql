--
-- ASYNC
--

--Should work. Send a valid message via a valid channel name
SELECT pg_notify('notify_async1','sample message1');
SELECT pg_notify('notify_async1','');
SELECT pg_notify('notify_async1',NULL);

-- Should fail. Send a valid message via an invalid channel name
SELECT pg_notify('','sample message1');
SELECT pg_notify(NULL,'sample message1');
SELECT pg_notify('notify_async_channel_name_too_long______________________________','sample_message1');

-- Should work. Valid NOTIFY/LISTEN/UNLISTEN commands
-- src/test/isolation/specs/async-notify-2.spec tests for actual usage.
NOTIFY notify_async2;
LISTEN notify_async2;
LISTEN SIMILAR TO 'notify_%';
UNLISTEN notify_async2;
UNLISTEN 'notify_%';
UNLISTEN 'notify_(%';
UNLISTEN *;

-- Should fail. Invalid LISTEN command
LISTEN *;
LISTEN notify_%;
LISTEN SIMILAR TO 'notify_(%';
LISTEN SIMILAR TO '*';

-- Should contain two listeners
LISTEN notify_async2;
LISTEN SIMILAR TO 'notify_async2';
LISTEN SIMILAR TO 'notify_%';
SELECT pg_listening_channels();

-- Should contain one listener
UNLISTEN 'notify_%';
SELECT pg_listening_channels();

-- Should not contain listeners
UNLISTEN *;
SELECT pg_listening_channels();

-- Should return zero while there are no pending notifications.
-- src/test/isolation/specs/async-notify.spec tests for actual usage.
SELECT pg_notification_queue_usage();
