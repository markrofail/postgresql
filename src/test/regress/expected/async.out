--
-- ASYNC
--
--Should work. Send a valid message via a valid channel name
SELECT pg_notify('notify_async1','sample message1');
 pg_notify 
-----------
 
(1 row)

SELECT pg_notify('notify_async1','');
 pg_notify 
-----------
 
(1 row)

SELECT pg_notify('notify_async1',NULL);
 pg_notify 
-----------
 
(1 row)

-- Should fail. Send a valid message via an invalid channel name
SELECT pg_notify('','sample message1');
ERROR:  channel name cannot be empty
SELECT pg_notify(NULL,'sample message1');
ERROR:  channel name cannot be empty
SELECT pg_notify('notify_async_channel_name_too_long______________________________','sample_message1');
ERROR:  channel name too long
-- Should work. Valid NOTIFY/LISTEN/UNLISTEN commands
-- src/test/isolation/specs/async-notify-2.spec tests for actual usage.
NOTIFY notify_async2;
LISTEN notify_async2;
LISTEN SIMILAR TO 'notify_%';
UNLISTEN notify_async2;
UNLISTEN 'notify_%';
UNLISTEN 'notify_(%';
UNLISTEN *;
-- Should fail. Invalid LISTEN command
LISTEN *;
ERROR:  syntax error at or near "*"
LINE 1: LISTEN *;
               ^
LISTEN notify_%;
ERROR:  syntax error at or near "%"
LINE 1: LISTEN notify_%;
                      ^
LISTEN SIMILAR TO 'notify_(%';
ERROR:  invalid regular expression: parentheses () not balanced
LISTEN SIMILAR TO '*';
ERROR:  invalid regular expression: quantifier operand invalid
-- Should contain two listeners
LISTEN notify_async2;
LISTEN SIMILAR TO 'notify_async2';
LISTEN SIMILAR TO 'notify_%';
SELECT pg_listening_channels();
 pg_listening_channels 
-----------------------
 notify_async2
 notify_%
(2 rows)

-- Should contain one listener
UNLISTEN 'notify_%';
SELECT pg_listening_channels();
 pg_listening_channels 
-----------------------
 notify_async2
(1 row)

-- Should not contain listeners
UNLISTEN *;
SELECT pg_listening_channels();
 pg_listening_channels 
-----------------------
(0 rows)

-- Should return zero while there are no pending notifications.
-- src/test/isolation/specs/async-notify.spec tests for actual usage.
SELECT pg_notification_queue_usage();
 pg_notification_queue_usage 
-----------------------------
                           0
(1 row)

